Metaverse/Source/Metaverse/Private/Components/AttributeComponent.cpp

```diff
@@ -0,0 +1,35 @@
+// Copyright 2022, Cronos Labs. All Rights Reserved
+
+#include "Components/AttributeComponent.h"
+
+// Sets default values for this component's properties
+UAttributeComponent::UAttributeComponent() {
+    // Set this component to be initialized when the game starts, and to be
+    // ticked every frame.  You can turn these features off to improve
+    // performance if you don't need them.
+    PrimaryComponentTick.bCanEverTick = true;
+
+    // ...
+}
+
+// Called when the game starts
+void UAttributeComponent::BeginPlay() {
+    Super::BeginPlay();
+
+    // ...
+}
+
+// Called every frame
+void UAttributeComponent::TickComponent(
+    float DeltaTime, ELevelTick TickType,
+    FActorComponentTickFunction *ThisTickFunction) {
+    Super::TickComponent(DeltaTime, TickType, ThisTickFunction);
+
+    // ...
+}
+
+void UAttributeComponent::SetCoin(float value) { Coin = value; }
+void UAttributeComponent::SetBalance(float value) { Balance = value; }
+void UAttributeComponent::AddCoin(float value) { Coin += value; }
```

Comment:
why is it empty?

Reply:
what do you mean empty?

Reply:
SetCoin function has no source code inside?

Reply:
Only one line.

Reply:
It updates the internal value, the UI is managed by HUD and overlay.